src/Avalonia.Controls/Application.cs

```diff
@@ -167,197 +160,28 @@ public IResourceDictionary Resources
         /// <inheritdoc/>
         IResourceNode IResourceNode.ResourceParent => null;
 
-        /// <summary>
-        /// Gets or sets the <see cref="ShutdownMode"/>. This property indicates whether the application is shutdown explicitly or implicitly. 
-        /// If <see cref="ShutdownMode"/> is set to OnExplicitShutdown the application is only closes if Shutdown is called.
-        /// The default is OnLastWindowClose
-        /// </summary>
-        /// <value>
-        /// The shutdown mode.
-        /// </value>
-        public ShutdownMode ShutdownMode { get; set; }
-
-        /// <summary>
-        /// Gets or sets the main window of the application.
-        /// </summary>
-        /// <value>
-        /// The main window.
-        /// </value>
-        public Window MainWindow { get; set; }
-
         /// <summary>
         /// Gets the open windows of the application.
         /// </summary>
         /// <value>
         /// The windows.
         /// </value>
         public WindowCollection Windows { get; }
```

Comment:
We should think about moving this back to `Window` itself or make this a member of `IClassicDesktopStyleApplicationLifetime`. If `Application.Current.ApplicationLifetime` is of that interface we can access this otherwise we don't need it anyways.

Reply:
I was thinking about adding `Created` and `Closed` routed events, so a class like `WindowsCollection` can be implemented without actually touching the `Window` code and without mandatory global state. Then we can make it an opt-in feature that is either enabled with `UseGlobalWindowsCollection()` while desktop lifetime keeps its own window list.

Reply:
I like the idea of driving it via events instead of it being a global list.